test/Semantics/resolve81.f90

```diff
@@ -0,0 +1,64 @@
+! RUN: %S/test_errors.sh %s %flang %t
+! C801 The same attr-spec shall not appear more than once in a given
+! type-declaration-stmt.
+!
+! R801 type-declaration-stmt ->
+!        declaration-type-spec [[, attr-spec]... ::] entity-decl-list
+!  attr-spec values are:
+!    PUBLIC, PRIVATE, ALLOCATABLE, ASYNCHRONOUS, CODIMENSION, CONTIGUOUS, 
+!    DIMENSION (array-spec), EXTERNAL, INTENT (intent-spec), INTRINSIC, 
+!    BIND(C), OPTIONAL, PARAMETER, POINTER, PROTECTED, SAVE, TARGET, VALUE, 
+!    VOLATILE
+module m
+
+  !ERROR: Attribute 'PUBLIC' cannot be used more than once
+  real, public, allocatable, public :: publicVar
+  !ERROR: Attribute 'PRIVATE' cannot be used more than once
+  real, private, allocatable, private :: privateVar
+  !ERROR: Attribute 'ALLOCATABLE' cannot be used more than once
+  real, allocatable, allocatable :: allocVar
+  !ERROR: Attribute 'ASYNCHRONOUS' cannot be used more than once
+  real, asynchronous, public, asynchronous :: asynchVar
+  !ERROR: Attribute 'CODIMENSION' cannot be used more than once
+  real, codimension[*], codimension[*] :: codimensionVar
+  !ERROR: Attribute 'CONTIGUOUS' cannot be used more than once
+  real, contiguous, pointer, contiguous :: contigVar(:)
+  !ERROR: Attribute 'DIMENSION' cannot be used more than once
+  real, dimension(5), dimension(5) :: arrayVar
+  !ERROR: Attribute 'EXTERNAL' cannot be used more than once
+  real, external, external :: externFunc
+  !ERROR: Attribute 'INTRINSIC' cannot be used more than once
+  real, intrinsic, bind(c), intrinsic :: cos
+  !ERROR: Attribute 'BIND(C)' cannot be used more than once
+  integer, bind(c), volatile, bind(c) :: bindVar
+  !ERROR: Attribute 'PARAMETER' cannot be used more than once
+  real, parameter, parameter :: realConst = 4.3
+  !ERROR: Attribute 'POINTER' cannot be used more than once
+  real, pointer, pointer :: realPtr
+  !ERROR: Attribute 'PROTECTED' cannot be used more than once
+  real, protected, protected :: realProt
+  !ERROR: Attribute 'SAVE' cannot be used more than once
+  real, save, save :: saveVar
+  !ERROR: Attribute 'TARGET' cannot be used more than once
+  real, target, target :: targetVar
+  !ERROR: Attribute 'VOLATILE' cannot be used more than once
+  real, volatile, volatile :: volatileVar
```

Comment:
There should be some tests where the duplicate attributes are not in the same statement.

Reply:
Wow, I didn't even realize that this was possible.  I'll add some tests.